giza_actions/model.py

```diff
@@ -79,12 +82,19 @@ def __init__(
             self._get_credentials()
             self.model = self._get_model(id)
             self.version = self._get_version(version)
-            self.session = self._set_session()
             self.framework = self.version.framework
             self.uri = self._retrieve_uri()
             self.endpoint_id = self._get_endpoint_id()
-            if output_path:
-                self._download_model(output_path)
+            self.session = self._set_session()
+            if output_path is not None:
+                self._output_path = output_path
+            else:
+                self._output_path = os.path.join(
+                    tempfile.gettempdir(),
+                    f"{self.model_id}_{self.version_id}_{self.model.name}",
+                )
+            self._download_model()
+            self._cache = Cache(os.path.join(os.getcwd(), "tmp", "cachedir"))
```

Comment:
Cache should be initialized before `_set_session` and `download_model` as if the user does not provide `output_path` set session will try to hit `self._cache` which has not been initialized. 

This is making the previous two existing tests to fail.